frontend/src/components/Searchbar/Searchbar.tsx

```diff
@@ -58,51 +73,113 @@ export interface SearchbarProps extends Omit<InputProps, 'onChange'> {
   value?: string
 
   /**
-   * Function to be invoked when the value in the searchbar input changes (but
-   * the search button has not been clicked).
+   * Optional. Function to be invoked when the value in the searchbar input
+   * changes (but the search button has not been clicked). Note: at least one of
+   * onSearch and onChange must be provided for the searchbar to work properly!
+   * @param newValue value of the search input
    */
   onChange?: (newValue: string) => void
+
+  /**
+   * Optional. If provided, adds a filter button to the searchbar. The initial
+   * (default) value of the filter, which is most likely the option for "No filter".
+   */
+  filterValue?: string
+
+  /**
+   * Optional. The remainder of the filter options, not including the default
+   * filterValue. Will be ignored unless filterValue has been provided.
+   */
+  filterOptions?: string[]
```

Comment:
hm this seems to break the convention of providing one prop specifying the current value and another prop specifying the full array of options (including the current value). see e.g. [`react-dropdown`](https://www.npmjs.com/package/react-dropdown). any particular reason why `filterValue` must not be an element of `filterOptions`?

Reply:
hmm, yeah. This was a decision that I did think about, and I went for this for two reasons. 

1. In this use case, default filter value will generally always be "no filter", so it made sense to view it not really as a filter option, but rather as the _name associated with_ "No filter". (for e.g. in the workspace context, "No filter" is called "All forms"). Then, filter options should be the list of options that represent the actual ways to filter whatever content you need. Perhaps in this situation the naming of `filterValue` is not great (maybe `noFilterName` or something like that?).
2. Asking users for an array along with a specified element of that array just seems funny to me. It's possible to also get rid of this by getting rid of `filterValue` and promising that `filterOptions[0]` will be the initial filter value, but then you lose the guarantee that `filterOptions` will be non-empty. Of course we can throw an error, but that also feels a bit inelegant i guess.

All that being said, I do see your point that the convention is just to provide the option array + an option from that array, and I don't think there's any issue with that (especially the example with react-dropdown). It would also make it more flexible, in the case that we want to initialize the filter with an option that isn't going to be shown as the first option. wdyt?

Reply:
personally I'd rather keep it unopinionated, so I wouldn't assume that you always want a "no filter" state (e.g. you might have a case where some filter is always active). my take is that we should stick with convention and just specify an element of the `filterOptions` array.

agree that it seems funny but specifying a `filterOptions` prop that doesn't actually include all the `filterOptions` is also funny. in this case I think it's best to stick to what devs are generally used to.

but will leave final decision up to you! it's our own component so we can always update it

Reply:
nah, i agree with you. I'll update it!

Reply:
[72a9d92](https://github.com/opengovsg/FormSG/pull/5458/commits/72a9d92bca2923427e1c5f8dbf187bf4b5bdb538)